src/Sarif.Sarifer.UnitTests/SpamBackgroundAnalyzerTests.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+using FluentAssertions;
+
+using Microsoft.CodeAnalysis.Sarif.Sarifer;
+
+using Moq;
+
+using Xunit;
+
+namespace Sarif.Sarifer.UnitTests
+{
+    public class SpamBackgroundAnalyzerTests
+    {
+        [Fact]
+        public void LoadPatternFiles_WhenDirectoryDoesNotExist_ShouldReturnEmptyList()
+        {
+            var mock = new Mock<IFileSystem>();
+            mock.Setup(fs => fs.DirectoryExists(It.IsAny<string>())).Returns(() => false);
+
+            List<SpamRule> rules = SpamBackgroundAnalyzer.LoadPatternFiles(mock.Object, Guid.NewGuid().ToString());
+
+            rules.Should().BeEmpty();
+        }
+
+        [Fact]
+        public void LoadPatternFiles_WhenDirectoryDoesExistButIsEmpty_ShouldReturnEmptyList()
+        {
+            var mock = new Mock<IFileSystem>();
```

Comment:
>mock [](start = 16, length = 4)

Nit: consider naming mocks more specifically, for example `mockFileSystem`.

That improves readability at the point of use:
```
List<SpamRule> rules = SpamBackgroundAnalyzer.LoadPatternFiles(mockFileSystem.Object, Guid.NewGuid().ToString());
```

Now it's clear to the reader that you're passing in a file system.

Reply:
I see this was merged. Consider applying these suggestions as part of the next PR.

---
In reply to: [532759638](https://github.com/microsoft/sarif-visualstudio-extension/pull/290#discussion_r532759638) [](ancestors = 532759638)